src/wooting-usb.c

```diff
@@ -63,108 +70,116 @@ static uint16_t getCrc16ccitt(const uint8_t *buffer, uint16_t size) {
 }
 
 typedef void (*set_meta_func)();
+void walk_hid_devices(hid_device* hid_info, set_meta_func meta_func);
 
 static void reset_meta(void) {
-  wooting_usb_meta.connected = false;
-
-  wooting_usb_meta.model = "N/A";
-  wooting_usb_meta.device_type = -1;
-  wooting_usb_meta.max_rows = 0;
-  wooting_usb_meta.max_columns = 0;
-  wooting_usb_meta.led_index_max = 0;
-  wooting_usb_meta.v2_interface = false;
-  wooting_usb_meta.layout = LAYOUT_UNKNOWN;
+  wooting_usb_meta->connected = false;
+
+  wooting_usb_meta->model = "N/A";
+  wooting_usb_meta->device_type = -1;
+  wooting_usb_meta->max_rows = 0;
+  wooting_usb_meta->max_columns = 0;
+  wooting_usb_meta->led_index_max = 0;
+  wooting_usb_meta->v2_interface = false;
+  wooting_usb_meta->layout = LAYOUT_UNKNOWN;
 }
 
 static void set_meta_wooting_one(void) {
-  wooting_usb_meta.model = "Wooting One";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD_TKL;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = WOOTING_ONE_RGB_COLS;
-  wooting_usb_meta.led_index_max = WOOTING_ONE_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = false;
+  wooting_usb_meta->model = "Wooting One";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD_TKL;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = WOOTING_ONE_RGB_COLS;
+  wooting_usb_meta->led_index_max = WOOTING_ONE_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = false;
 }
 
 static void set_meta_wooting_one_v2(void) {
   set_meta_wooting_one();
-  wooting_usb_meta.v2_interface = true;
+  wooting_usb_meta->v2_interface = true;
 }
 
 static void set_meta_wooting_two(void) {
-  wooting_usb_meta.model = "Wooting Two";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = WOOTING_TWO_RGB_COLS;
-  wooting_usb_meta.led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = false;
+  wooting_usb_meta->model = "Wooting Two";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = WOOTING_TWO_RGB_COLS;
+  wooting_usb_meta->led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = false;
 }
```

Comment:
Wouldn't it be better if all these meta-functions accepted the pointer as an argument and updated that rather than modifying the global variable?

Reply:
but that's basically the same right? the globally stored pointer gets loaded from an array when switching selected device. so passing it in or not makes no real difference from what I can tell. although im also not a C master